hail/python/hail/expr/aggregators/aggregators.py

```diff
@@ -486,19 +486,54 @@ def all(condition) -> BooleanExpression:
     return count_where(~condition) == 0
 
 
-@typecheck(expr=expr_any)
-def counter(expr) -> DictExpression:
+@typecheck(expr=expr_any, weight=nullable(expr_any))
+def counter(expr, weight=None) -> DictExpression:
```

Comment:
I would prefer if weight is a keyword-only argument, so it's totally clear what's going on when this is used (this makes it easier to maintain back-compatibility in interfaces too).

The typecheck should also be expr_int64

Reply:
It is now keyword-only. The type check is now `expr_numeric`. Floats are fine. Added a test demonstrating this.